DB/src/main/java/io/deephaven/stream/StreamToTableAdapter.java

```diff
@@ -228,9 +247,37 @@ public DynamicTable table() {
 
     @Override
     public void close() {
-        log.info().append("Deregistering ").append(StreamToTableAdapter.class.getSimpleName()).append('-').append(name)
-                .endl();
-        liveTableRegistrar.removeTable(this);
+        if (!alive) {
+            return;
+        }
+        synchronized (this) {
+            if (!alive) {
+                return;
+            }
+            alive = false;
+            log.info().append("Deregistering ").append(StreamToTableAdapter.class.getSimpleName()).append('-').append(name)
+                    .endl();
+            liveTableRegistrar.removeTable(this);
+            final Runnable localShutdownCallback = shutdownCallback;
+            if (localShutdownCallback != null) {
+                localShutdownCallback.run();
+            }
+        }
+        // CODE REVIEW DISCUSSION: SHOULD WE ATTEMPT THIS CLEANUP? I THINK NOT.
```

Comment:
I think leaving the switch column sources alone makes sense; because we don't want to change the data of the table off the LTM refresh loop (or without our singularly allowed notification).  We could throw out our bufferChunkSources though.

Reply:
I was worried they might be in use concurrently. Do you think that's a non-issue?

Reply:
So the buffered sources are captured under a sync block; but are filled before that.  We could solve that by making a final refresh release resources if we are done; but then we can't deregister.  It is really a question of whether we care to release these things back to the pool, and if we actually had unit/component tests that could execute w/ chunk tracking then we would blow up.

Reply:
I had decided that intentionally leaking the chunks and letting GC have them was "fine" in this case, since it's an occasional occurrence on shutdown and not a per-operation thing.
Either way we're leaking some chunks, since there will be some in the switch sources.